nautilus_trader/adapters/okx/data.py

```diff
@@ -377,46 +377,53 @@ async def _request_trade_ticks(self, request: RequestTradeTicks) -> None:
     async def _request_bars(self, request: RequestBars) -> None:
         start = request.start
         end = request.end
-
-        self._log.warning(
-            f"Bar requests time range not implemented: start={start}, end={end}, using no time range",
-        )
-        # TODO: Implement correct time range requests
-        start = None
-        end = None
-
-        # Validate and clamp limit parameter
         limit = request.limit
-        if limit is not None and limit > 0:
-            if limit > 300:
-                self._log.warning(
-                    f"Requested limit {limit} exceeds OKX maximum of 300, "
-                    f"clamping to 300 for {request.bar_type}",
-                )
-            limit = min(limit, 300)
-        else:
-            limit = 100  # Default limit
-
-        pyo3_bar_type = nautilus_pyo3.BarType.from_str(str(request.bar_type))
-
-        pyo3_bars = await self._http_client.request_bars(
-            bar_type=pyo3_bar_type,
-            start=ensure_pydatetime_utc(start),
-            end=ensure_pydatetime_utc(end),
-            limit=limit,
-        )
-        bars = Bar.from_pyo3_list(pyo3_bars)
 
-        self._handle_bars(
-            request.bar_type,
-            bars,
-            None,
-            request.id,
-            request.start,
-            request.end,
-            request.params,
+        self._log.debug(
+            f"Requesting bars: bar_type={request.bar_type}, start={start}, end={end}, "
+            f"limit={limit}",
         )
 
+        try:
+            pyo3_bar_type = nautilus_pyo3.BarType.from_str(str(request.bar_type))
+
+            # The Rust implementation now handles:
+            # - Endpoint selection logic (regular vs history based on 100-day boundary)
+            # - Limit validation (300 for regular endpoint, 100 for history endpoint)
+            # - Time range validation
+            # - Timezone handling
+            pyo3_bars = await self._http_client.request_bars(
+                bar_type=pyo3_bar_type,
+                start=ensure_pydatetime_utc(start),
+                end=ensure_pydatetime_utc(end),
+                limit=limit,
+            )
+            bars = Bar.from_pyo3_list(pyo3_bars)
+
+            self._handle_bars(
+                request.bar_type,
+                bars,
+                None,
+                request.id,
+                request.start,
+                request.end,
+                request.params,
+            )
+        except Exception as e:
```

Comment:
Lets remove the exceptions handling, it's standard to allow any exception to bubble up into the data clients own exception handling with logging.